vermouth/processors/apply_rubber_band.py

```diff
@@ -365,7 +365,43 @@ def same_chain(graph, left, right):
     node_right = graph.nodes[right]
     return node_left.get('chain') == node_right.get('chain')
 
+def same_region(graph, left, right, regions):
```

Comment:
I disagree with this signature since other "criterions" (such as `same_chain`) only accept `graph`, `left` and `right`. I think the best option is to either create a SameRegion class with a `__call__` method, or a function factory:
```py
def make_same_region_criterion(regions):
    regions = copy.deepcopy(regions)
    def same_region(graph, left, right):
        ...
    return same_region
```
You can then do something like this:
```py
criterion = make_same_region_criterion(args.regions)
```
This `criterion` is then a function that has the same signature as `same_chain`. This would also save you from having to pass `regions` to all functions here.